### ISSUE TITLE
Slow Click with KeepKey on continue click with yearn vaults

### ISSUE THREAD

**Issue by tshifty** [shapeshift/web#940]
Description: When on the deposit page for yearn vaults there is a lag between clicking continue, one you wait a minute or two it works but will not if you go directly to the yearn vaults after a refresh.

Action Items:
- If out what is causing this delayed response time with KK and yearn vaults
- Once the issue has been found make a PR of the fix and reach out to Operations to test your PR

**Comment by gomesalexandre** [comment]
The issue is that
- we have currently many (59) `<EarnOpportunityRow />` rows, each of them awaiting a `await chainAdapter.getAddress({ wallet })`  promise
- `<YearnDeposit />` fires a single similar request, but given how promises and the main thread work, this one resolves after all the opportunity rows requests resolve. We have an early return on button click - ` if (!state.userAddress) return` - that kicks in because of this.

This could be solved by not awaiting `getAddress()` on every row but only once on `<EarnOpportunities />` then passing it down, but it will require getting the `chain` there instead of in `vault`. We can use the `caip19` for that, in particular `fromCaip19`.

**Comment by 0xean** [comment]
@gomesalexandre - to handle it at some point, low priority.

**Comment by gomesalexandre** [comment]
Looks like https://github.com/shapeshift/web/commit/860bd99e3df0cb4f6c9af9b94721db842932bc34 from @DaoDev44 fixed it, I'm not able to reproduce it anymore

**Comment by tshifty** [comment]
Confirmed fixed. Thanks for checking @gomesalexandre
